File: client/src/services/UserService.js
import axios from "axios";

const QUIZ_BASE = "http://192.168.0.9:8080/user";

class UserService {
    getAllUsers() {
        return axios.get(QUIZ_BASE);
    }
    addUser(userData) {
        return axios.post(QUIZ_BASE, userData);
    }
}   

export default new UserService();

File: server/src/main/java/com/quiz/services/UserService.java
package com.quiz.services;

import com.quiz.exceptions.AlreadyExistException;
import com.quiz.model.User;
import com.quiz.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.RequestBody;

import java.util.List;

@Service
public class UserService {

    private final UserRepository userRepository;

    @Autowired
    public UserService (UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public User addUser(@RequestBody User user) {
        if(!userRepository.existsByUserName(user.getUserName())) {
            return userRepository.save(user);
        } throw new AlreadyExistException("Username is already in use. Please choose another name.");
    }

    public List<User> getAllUsers() {
        return userRepository.findAll();
    }
}
